[Cheah Yan (Xie Yan)] iP

CheahYan opened the pull request:
**DukeLite**
> "Your mind is for having ideas, not holding them." -David Allen ([source](https://www.goodreads.com/quotes/348103-your-mind-is-for-having-ideas-not-holding-them#:~:text=Quote%20by%20David%20Allen%3A%20%E2%80%9CYour,ideas%2C%20not%20holding%20them.%E2%80%9D))

DukeLite frees your mind of having to remember things you need to do. It's,

* text-based
* easy to learn
* ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://github.com/CheahYan/ip/releases)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

CheahYan committed:
Level-1

CheahYan committed:
Level-2

CheahYan committed:
Level-3

CheahYan committed:
Level-4

CheahYan committed:
Level-5

CheahYan committed:
Level-6

CheahYan committed:
ui testing

CheahYan committed:
Level-7

CheahYan committed:
Level-7 update

CheahYan committed:
Level-8

CheahYan committed:
Revert "Level-8"

This reverts commit 881bfc16cd91490b08c5b4dd562fc04a199646d3.

CheahYan committed:
Merge branch 'branch-Level-7'

CheahYan committed:
Level-8

CheahYan committed:
debugging after merge conflicts

darrenhoon commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,22 @@
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
```
Perhaps a class should have a header comment to inform users about what the class does. Same suggestion for the other .java files

darrenhoon commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,22 @@
+
+    protected LocalDateTime date;
+
+    public Deadline(String description, String date) {
```
Perhaps a method could have a header comment to inform users about what the method does.

Same suggestion for the other methods both in this file and other .java files

darrenhoon commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,15 @@
+package duke;
+
+public class DukeException extends Exception {
```
Great work using inheritance! LGTM!

darrenhoon commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,28 @@
+        this.isDone = false;
+    }
+
+    public String convertToFile() {
```
Great use of naming conventions!

g4ryy replied:
Perhaps a more suitable name is getDescription.  I feel that naming it convertToFile can be misleading since this method does not handle any conversion.

g4ryy commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,22 @@
+    public Deadline(String description, String date) {
+        super(description);
+
+        DateTimeFormatter scanned = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
```
Perhaps a more meaningful name for this variable would be datePattern or dateFormat.

darrenhoon commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,213 @@
+    protected static final String LOCAL_FILE = "data/duke.txt";
+
+    public static void appendToFile(String filePath, String textToAppend) throws IOException {
+        FileWriter fw = new FileWriter(filePath, true); // create a FileWriter in append mode
```
leaving comments to help readers enables a better understanding of your code and might even help to hasten the onboarding process. Good job!

darrenhoon reviewed (commented):
Overall, the use of good naming conventions and meaningful variable names shows that you placed a lot of thought into how your files can be easily understood by others.  Having comment headers would greatly enhance your work.

Overall, LGTM!

g4ryy commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,213 @@
+
+        Scanner sc = new Scanner(System.in);
+        String a = sc.nextLine();
+        String[] b = a.split(" ", 2);
```
Perhaps it would be good if you could avoid using single letters to name variables unless they are iterator variables. I noticed this issue in other parts too.

g4ryy commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,213 @@
+
+                } else if (details[0].equals("delete")) {
+                    int taskIndex = Integer.valueOf(details[1]);
+                    Task removed = history.get(taskIndex - 1);
```
Perhaps a more meaningful name would be removedTask to indicate that this is the task that had been removed.

g4ryy commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,213 @@
+            sc.close();
+
+    }
+}
```
Perhaps you could apply more OOP by extracting out closely related code as classes, rather than letting the Main method handle everything.

g4ryy reviewed (commented):
Overall, your code is structured logically which makes it easy to follow. I think the main issue to fix would the naming of a few variables and methods where it can be misleading or difficult to understand. Also, it would be great to include Javadoc comments which I believe would improve your code's readability. I hope my reviews will be useful to you! 🙂

CheahYan committed:
OOP

CheahYan committed:
oop updates

CheahYan committed:
A-MoreOOP

houtenteo commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,56 @@
+        System.out.println("Bye! Hope to see you again soon!");
+    }
+
+    public void list(TaskList tasklist) {
```
I believe the coding standards suggest that method names should be verbs. In this case, 'list' could be a noun. Maybe a better option would be to name the method 'listAll'.

houtenteo commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,28 @@
+        return (isDone ? "X" : " ");
+    }
+
+    public void Done() {
```
Method names should be in camel case.

houtenteo commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,104 @@
+        return this.twoPart;
+    }
+
+    public boolean isBye() {
```
I guess u can shorten this into one-liner by just returning 'this.command.equals("bye"')'. Helps improve readability somewhat but I guess I'm just nitpicking here.

houtenteo commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,104 @@
+        return this.twoPart[0];
+    }
+
+    public int secondPartInInt() throws DukeException {
```
Method names should be verbs. So this would be better named as 'getSecondPartInInt'.

houtenteo reviewed (commented):
Great work overall :)

CheahYan committed:
updates

CheahYan committed:
A-JUnit

CheahYan committed:
A-Jar

CheahYan committed:
A-JavaDoc

CheahYan committed:
A-CodingStandard

CheahYan committed:
Level-9

CheahYan committed:
Merge branch 'branch-A-JavaDoc'

CheahYan committed:
Merge branch 'branch-Level-9'

CheahYan committed:
Merge remote-tracking branch 'origin/add-gradle-support'

CheahYan committed:
Gradle and CheckStyle

CheahYan committed:
not sure whats going on

CheahYan committed:
idk

CheahYan committed:
idk

CheahYan committed:
idk

CheahYan committed:
idk

CheahYan committed:
WHAT IS GOING ON

CheahYan committed:
WHAT IS GOING ON

CheahYan committed:
WHAT IS GOING ON

CheahYan committed:
debugging

CheahYan committed:
Merge branch 'master' of https://github.com/CheahYan/ip

CheahYan committed:
updates

CheahYan committed:
Merge branch 'master' of https://github.com/CheahYan/ip

CheahYan committed:
updates

CheahYan committed:
Merge branch 'reset-to-gradle'

CheahYan committed:
updates

CheahYan committed:
Level-7 update

CheahYan committed:
GUI

CheahYan committed:
debug

CheahYan committed:
level 10

CheahYan committed:
update

CheahYan committed:
Update DialogBox.java

CheahYan committed:
update

CheahYan committed:
Add assertions

Program does not have Assertions

Assertions help to indicate a possible bug in the code at runtime.

Adding assertions will allow us to more easily identify mistakes that we programmers make in the code.

CheahYan committed:
Updates to GUI

GUI does not display the full TaskList if there is a large number of tasks.

Users will not be able to properly view the entire list of tasks easily

DialogBox.fxml file changed allows us to easily view all the tasks in the list when "list" command is called.

CheahYan committed:
Improve Code Quality to all files

Current code is not very readable for other programmers.

This can be very hard for programmers to improve upon the code in the future.

Readability of code is improved with current commit

CheahYan committed:
Merge branch 'master' into branch-A-Assertions

CheahYan committed:
Merge pull request #2 from CheahYan/branch-A-Assertions

Add assertions

CheahYan committed:
Merge branch 'master' into branch-A-CodeQuality

CheahYan committed:
Merge pull request #3 from CheahYan/branch-A-CodeQuality

Improve Code Quality to all files

CheahYan committed:
B-Reminders

Added Reminder feature to Duke.

CheahYan committed:
Add Ui.png and updated MainWindow.fxml

CheahYan committed:
Set theme jekyll-theme-cayman

CheahYan committed:
updates to code

CheahYan committed:
Merge branch 'master' of https://github.com/CheahYan/ip

CheahYan committed:
A-UserGuide

CheahYan committed:
more SLAP for Storage
